query-engine/core/src/telemetry/capturing/capturer.rs

```diff
@@ -118,32 +99,7 @@ impl SpanProcessor for Processor {
     /// In the case of mongo, an event represents the query, but it needs to be transformed before
     /// capturing it. `Event::query_event` does that.    
     fn on_end(&self, span_data: SpanData) {
-        let trace_id = span_data.span_context.trace_id();
-
-        let mut locked_storage = self.storage.lock().unwrap();
-        if let Some(storage) = locked_storage.get_mut(&trace_id) {
-            let settings = storage.settings.clone();
-
-            let (events, span) = models::TraceSpan::from(span_data).split_events();
-
-            if settings.traces_enabled() {
-                storage.traces.push(span);
-            }
-
-            if storage.settings.logs_enabled() {
-                events.into_iter().for_each(|log| {
-                    let candidate = Candidate {
-                        value: log,
-                        settings: &settings,
-                    };
-                    if candidate.is_loggable_query_event() {
-                        storage.logs.push(candidate.query_event())
-                    } else if candidate.is_loggable_event() {
-                        storage.logs.push(candidate.value)
-                    }
-                });
-            }
-        }
+        _ = task::span_data_processed(span_data);
```

Comment:
We should change this to an `.unwrap()`, or if we really want to fail silently an `.ok()` on the result. The reason is that the discard pattern (`_`) silences `#[must_use]` annotations. The most common types with a `#[must_use]` are `Result` and `Future`.

It's common to change that kind of functions from sync to async (and vice versa). If we did that:

- With the discard pattern (`_`), the future is never polled, and nothing will tell you about it, except that the code in your future is silently not executing.
- With `.ok()` or `.unwrap()`, you get a compiler error, because these methods don't exist on futures. Then you can change that to `.await.ok()` and call it a day.

It probably sounds pedantic, but I've had this exact thing happen to me multiple times :p

Reply:
It doesn't sound pedantic, this kind of idioms are very hard to realize without experience and are very illustrative. Thank you! 🙇🏾